docs/user-guide/runtime/configuration.md

```diff
@@ -16,103 +17,219 @@ When running as an [embedded application](embedding), the configuration options
 are passed in the call to `RED.init()`. However, when run in this mode, certain
 properties are ignored and are left to the embedding application to implement.
 
-### Runtime Configuration
+The settings are split into the following sections:
+- Flow File and User Directory
+- Security
+- Server 
+- Runtime
+- Editor
+- Node
+ 
+### Flow File and User Directory
 
-flowFile
-: the file used to store the flows. Default: `flows_<hostname>.json`
+**flowFile**
+: the file used to store the flows. If not set, defaults to `flows_<hostname>.json`<br /> Default: `flowFile: 'flows.json',`
 
-userDir
-: the directory to store all user data, such as flow and credential files and all
-  library data. Default: `$HOME/.node-red`
+**credentialSecret**
+: By default, credentials are encrypted in storage using a generated key. To specify your own secret, set the following property. If you want to disable encryption of credentials, set this property to false. Note: once you set this property, do not change it - doing so will prevent node-red from being able to decrypt your existing credentials and they will be lost. Example: `credentialSecret: "a-secret-key",`
 
-nodesDir
+**flowFilePretty**
+:  By default, the flow JSON will be formatted over multiple lines making
+     it easier to compare changes when using version control.
+     To disable pretty-printing of the JSON set the following property to false. Default: `flowFilePretty: true`
+
+
+**userDir**
+: By default, all user data is stored in a directory called `.node-red` under the user's home directory. To use a different location, the following property can be used `userDir: '/home/nol/.node-red/',` Default: `$HOME/.node-red`
+
+**nodesDir**
+*Since Node-RED 3.0.0.*
 : a directory to search for additional installed nodes. Node-RED searches the `nodes`
   directory under the *userDir* directory. This property allows an additional directory
   to be searched, so that nodes can be installed outside of the Node-RED install
   structure. Default: `$HOME/.node-red/nodes`
```

Comment:
```suggestion
**nodesDir**
: This property allows an additional directory to be searched for nodes to import.

*Before Node-RED 3.0.0*
: This setting can be a single path `string` only

*Since Node-RED 3.0.0*
: This setting can be...
* A single path `string`
* An array of  path `string`s
```

I think I forgot to merge the updated comments for `nodesDir` when i raised the PR (BAH!) 
the above (or if you can think of better words for "it can be a string or array of strings") is more up-to-date.

Reply:
`nodesDir`  is a good example of some settings options that have changed with added features from version to version. If we add how it was Before and after v3 it could start getting messy (can of worms). Personally i think the docs should inform about the configuration as it stands now with the latest Node-red version. and add to the note at the top : "*Please note that some of the options may not apply for older versions of Node-red. (**consult the comments in your settings.js file**)*"